### ISSUE TITLE
subscribe/unsubscribe/publish

### ISSUE THREAD

**Issue by wpbloger** [phpredis/phpredis#208]
Hello,
 I have does not work function subscribe (subscription channels), when I run the code from the example on eg. github, the server for a long time boot and then returns a 504 error, it seems to the eternal cycle. 
Tell me what am I doing wrong?
 And another question, how do I unsubscribe from a channel? 
Version Redis: 3.4
phpredis just reinstalled. 
The code which I'm testing:
$redis = new Redis();
$redis->connect('127.0.0.1',6379);
function f($redis, $chan, $msg) {
    switch($chan) {
        case 'chan-1':
            //...
            break;

```
    case 'chan-2':
        //...
        break;

    case 'chan-2':
        //...
        break;
}
```

}
$redis->subscribe(array('chan-1', 'chan-2', 'chan-3'), 'f'); // subscribe to 3 chans

**Comment by jmshelby** [comment]
I'm still curious about your second question: `how do I unsubscribe from a channel?`

I've tried using the unsubscribe method on the redis connection, which does issue unsubscribe to redis itself, however the php process stays in a blocked state. How can I get the process out of it's 'blocked' state? For example, can I call a method on the redis connection inside the handler function that will return the process back from the subscribe function call?

**Comment by malaysarkar** [comment]
The best way I found so far to unblock the "subscribe function call" is to close the redis connection by calling $redis->close(); But you won't be able to return a value though.

Additional info: I also noticed; using $redis->subscribe($ch, $foo); multiple time from withing the callback keeps blocking the thread. Thus keeps increasing the call-stack. Better to use $redis->rawCommand("subscribe",$c); if you want additional channels to be included later.

**Comment by yatsukhnenko** [comment]
Old issue without any activity for a long time.

**Comment by boussou** [comment]
So it means, when there is no solution found, you decide there is no problem

**Comment by boussou** [comment]
at least try this in php.ini: 

; Default timeout for socket based streams (seconds)
; http://php.net/default-socket-timeout
default_socket_timeout = 86400

**Comment by yatsukhnenko** [comment]
@boussou when I was coming to this project I closed issues without activity to concentrate on actual problems. If problem exists in current stable version please open new issue.

**Comment by boussou** [comment]
The solution is common to all versions. it is a socket problem, solved by a longer timeout.
when the timeout is reached, don't try to reconnect, kill the process & restart.
